packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -329,6 +342,72 @@ class ToolbarOptions {
   final bool selectAll;
 }
 
+/// Content insertion configuration for [EditableText].
+///
+/// The configuration provides a handler for any rich content inserted through
+/// the system input method, and also provides the ability to limit the mime
+/// types of the inserted content.
+///
+/// By default, [EditableText] has no content insertion configuration set.
+class ContentInsertionConfiguration {
+  /// Creates a content insertion configuration with the specified options.
+  ///
+  /// A handler for inserted content, in the form of [onContentInserted], must
+  /// be supplied.
+  ///
+  /// The allowable mime types of inserted content may also
+  /// be provided via [contentInsertionMimeTypes], which cannot be an empty list.
+  ContentInsertionConfiguration({
+    required this.onContentInserted,
+    this.contentInsertionMimeTypes = kDefaultContentInsertionMimeTypes,
+  }) : assert(onContentInserted != null),
+       assert(contentInsertionMimeTypes != null),
+       assert(contentInsertionMimeTypes.isNotEmpty);
```

Comment:
These parameters and assertions make a lot of sense compared to how we were doing it on EditableText before :+1: